ddp-workspace/projects/ddp-sdk/src/lib/components/address/addressEmbedded.component.ts

```diff
@@ -225,59 +234,74 @@ export class AddressEmbeddedComponent implements OnDestroy, OnInit {
-      take(1),
-      tap(() => busyCounter$.next(1)),
-      mergeMap((state) => this.addressService.findDefaultAddress().pipe(
-        map(defaultAddress => [state, defaultAddress]))
-      ),
-      tap(([state, defaultAddress]) =>
-        defaultAddress && this.stateUpdates$.next({ inputAddress: defaultAddress as Address })),
-      // filter for case where we need to go on to look for a temp address?
-      filter(([state, defaultAddress]) => !defaultAddress && !!(state as ComponentState).activityInstanceGuid),
-      map(([state, _]) => state as ComponentState),
-      mergeMap((state) => this.addressService.getTempAddress(state.activityInstanceGuid)),
-      tap((tempAddress) => tempAddress && this.stateUpdates$.next({ inputAddress: tempAddress as Address })),
-      // fake that the address was just entered. Perhaps this can become a separate subject?
-      // guess we are saving temp address again. No harm but not nice either.
-      tap((tempAddress) => this.inputComponentAddress$.next(tempAddress)),
-      finalize(() => busyCounter$.next(-1))
+        take(1),
+        tap(() => busyCounter$.next(1)),
+        // let's disable form while we try to load initial address
+        tap(() => this.stateUpdates$.next({isTemporarilyDisabled: true})),
+        mergeMap((state) => this.addressService.findDefaultAddress().pipe(
+            map(defaultAddress => [state, defaultAddress]))
+        ),
+        tap(([_, defaultAddress]: [ComponentState, Address | null]) =>
+            defaultAddress && this.inputAddress$.next(defaultAddress)),
+        // filter for case where we need to go on to look for a temp address?
+        filter(([state, defaultAddress]) => !defaultAddress && !!(state as ComponentState).activityInstanceGuid),
+        // map(([state, _]) => state as ComponentState),
+        mergeMap(([state, _]) => this.addressService.getTempAddress(state.activityInstanceGuid)),
+        tap((tempAddress) => this.inputAddress$.next(tempAddress)),
+        // fake that the address was just entered. Perhaps this can become a separate subject?
+        // guess we are saving temp address again. No harm but not nice either.
+        tap((tempAddress) => this.inputComponentAddress$.next(tempAddress)),
+        finalize(() => {
+            this.stateUpdates$.next({isTemporarilyDisabled: false});
+            busyCounter$.next(-1);
+        }),
     );
 
-    // derived observables
-    this.isReadOnly$ = this.state$.pipe(
-      pluck('isReadOnly'),
-      distinctUntilChanged(),
-      shareReplay()
+    this.staticCountry$ = this.inputAddress$.pipe(
+        map(address => {
+          // ok to look at country at ngOnInit. We don't want to do this more than once
+          if ((this.country && (!(address) || (address.country === this.country)))) {
+            return this.country;
+          } else {
+            return null;
+          }
+        }),
+        distinctUntilChanged(),
+        shareReplay()
```

Comment:
If there is no country passed, then we do everything as before. If we have one, before we configure a "staticCountry" we want to make sure that the address does not have a country that differs.  Want to be able to edit country if address has a different country.